File: Ej 2/factoria_carretera.py
from factoria_carrera_bicicleta import FactoriaCarreraYBicicleta
from carrera_carretera import CarreraCarretera
from bicicleta_carretera import BicicletaCarretera
import random
    
class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def __init__(self):
        self.prototype_bicicleta = BicicletaCarretera(ID=0)
        self.prototype_carrera = CarreraCarretera()
    
    def crearCarrera(self):
        return self.prototype_carrera.clonar()
    
    def crearBicicleta(self):
        clone = self.prototype_bicicleta.clonar()
        clone.ID = random.randint(0, 10000)
        return clone
    

File: Practica1/Ej 1/src/isp1/FactoriaCarretera.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package isp1;
import java.util.Random;

/**
 *
 * @author joy11
 */
public class FactoriaCarretera implements FactoriaCarreraYBicicleta {
    
    @Override
    public Carrera crearCarrera(){
        return new CarreraCarretera();
    };
    
    @Override
    public Bicicleta crearBicicleta(){
        BicicletaCarretera bici = new BicicletaCarretera();
        Random r = new Random();
        bici.ID = r.nextInt();
        
        return bici;
    };
    
}
